// @nexus:react/components.js
/* eslint-disable */
/* tslint:disable */
/* auto-generated react proxies */
import { createReactComponent } from './react-component-lib';
import { applyPolyfills, defineCustomElements } from '@nexus/core/loader';
applyPolyfills().then(() => defineCustomElements());
export const NexusAccordion = /*@__PURE__*/ createReactComponent('nexus-accordion');
export const NexusAccordionContent = /*@__PURE__*/ createReactComponent('nexus-accordion-content');
export const NexusAccordionGroup = /*@__PURE__*/ createReactComponent('nexus-accordion-group');
export const NexusAccordionTrigger = /*@__PURE__*/ createReactComponent('nexus-accordion-trigger');
export const NexusAvatar = /*@__PURE__*/ createReactComponent('nexus-avatar');
export const NexusBadge = /*@__PURE__*/ createReactComponent('nexus-badge');
export const NexusBarChart = /*@__PURE__*/ createReactComponent('nexus-bar-chart');
export const NexusBreadcrumb = /*@__PURE__*/ createReactComponent('nexus-breadcrumb');
export const NexusBreadcrumbItem = /*@__PURE__*/ createReactComponent('nexus-breadcrumb-item');
export const NexusCard = /*@__PURE__*/ createReactComponent('nexus-card');
export const NexusCardAvatar = /*@__PURE__*/ createReactComponent('nexus-card-avatar');
export const NexusCardBody = /*@__PURE__*/ createReactComponent('nexus-card-body');
export const NexusCardBodyImage = /*@__PURE__*/ createReactComponent('nexus-card-body-image');
export const NexusCardFooter = /*@__PURE__*/ createReactComponent('nexus-card-footer');
export const NexusCardHeader = /*@__PURE__*/ createReactComponent('nexus-card-header');
export const NexusCardHeaderButton = /*@__PURE__*/ createReactComponent('nexus-card-header-button');
export const NexusCardHeaderDetails = /*@__PURE__*/ createReactComponent('nexus-card-header-details');
export const NexusCardHeaderSubtitle = /*@__PURE__*/ createReactComponent('nexus-card-header-subtitle');
export const NexusCardHeaderTitle = /*@__PURE__*/ createReactComponent('nexus-card-header-title');
export const NexusCarousel = /*@__PURE__*/ createReactComponent('nexus-carousel');
export const NexusCarouselItem = /*@__PURE__*/ createReactComponent('nexus-carousel-item');
export const NexusCharts = /*@__PURE__*/ createReactComponent('nexus-charts');
export const NexusCheckbox = /*@__PURE__*/ createReactComponent('nexus-checkbox');
export const NexusChip = /*@__PURE__*/ createReactComponent('nexus-chip');
export const NexusChipGroup = /*@__PURE__*/ createReactComponent('nexus-chip-group');
export const NexusDropdown = /*@__PURE__*/ createReactComponent('nexus-dropdown');
export const NexusErrorMessage = /*@__PURE__*/ createReactComponent('nexus-error-message');
export const NexusFileUpload = /*@__PURE__*/ createReactComponent('nexus-file-upload');
export const NexusFooter = /*@__PURE__*/ createReactComponent('nexus-footer');
export const NexusFooterBottom = /*@__PURE__*/ createReactComponent('nexus-footer-bottom');
export const NexusFooterColumn = /*@__PURE__*/ createReactComponent('nexus-footer-column');
export const NexusFooterContent = /*@__PURE__*/ createReactComponent('nexus-footer-content');
export const NexusFooterContentTitle = /*@__PURE__*/ createReactComponent('nexus-footer-content-title');
export const NexusFooterLogo = /*@__PURE__*/ createReactComponent('nexus-footer-logo');
export const NexusFormField = /*@__PURE__*/ createReactComponent('nexus-form-field');
export const NexusGaugeChart = /*@__PURE__*/ createReactComponent('nexus-gauge-chart');
export const NexusHeader = /*@__PURE__*/ createReactComponent('nexus-header');
export const NexusHeaderLogo = /*@__PURE__*/ createReactComponent('nexus-header-logo');
export const NexusHeaderMenu = /*@__PURE__*/ createReactComponent('nexus-header-menu');
export const NexusIcon = /*@__PURE__*/ createReactComponent('nexus-icon');
export const NexusInput = /*@__PURE__*/ createReactComponent('nexus-input');
export const NexusLabel = /*@__PURE__*/ createReactComponent('nexus-label');
export const NexusLineChart = /*@__PURE__*/ createReactComponent('nexus-line-chart');
export const NexusLoader = /*@__PURE__*/ createReactComponent('nexus-loader');
export const NexusMenu = /*@__PURE__*/ createReactComponent('nexus-menu');
export const NexusMenuItem = /*@__PURE__*/ createReactComponent('nexus-menu-item');
export const NexusMessage = /*@__PURE__*/ createReactComponent('nexus-message');
export const NexusModal = /*@__PURE__*/ createReactComponent('nexus-modal');
export const NexusModalBody = /*@__PURE__*/ createReactComponent('nexus-modal-body');
export const NexusModalFooter = /*@__PURE__*/ createReactComponent('nexus-modal-footer');
export const NexusModalHeader = /*@__PURE__*/ createReactComponent('nexus-modal-header');
export const NexusModalHeaderSubtitle = /*@__PURE__*/ createReactComponent('nexus-modal-header-subtitle');
export const NexusNotification = /*@__PURE__*/ createReactComponent('nexus-notification');
export const NexusOption = /*@__PURE__*/ createReactComponent('nexus-option');
export const NexusPagination = /*@__PURE__*/ createReactComponent('nexus-pagination');
export const NexusPieChart = /*@__PURE__*/ createReactComponent('nexus-pie-chart');
export const NexusPinEntry = /*@__PURE__*/ createReactComponent('nexus-pin-entry');
export const NexusProgressBar = /*@__PURE__*/ createReactComponent('nexus-progress-bar');
export const NexusProgressBarLabel = /*@__PURE__*/ createReactComponent('nexus-progress-bar-label');
export const NexusRadio = /*@__PURE__*/ createReactComponent('nexus-radio');
export const NexusSearch = /*@__PURE__*/ createReactComponent('nexus-search');
export const NexusSearchList = /*@__PURE__*/ createReactComponent('nexus-search-list');
export const NexusSearchListItem = /*@__PURE__*/ createReactComponent('nexus-search-list-item');
export const NexusSelect = /*@__PURE__*/ createReactComponent('nexus-select');
export const NexusSlider = /*@__PURE__*/ createReactComponent('nexus-slider');
export const NexusStackedBarChart = /*@__PURE__*/ createReactComponent('nexus-stacked-bar-chart');
export const NexusStepper = /*@__PURE__*/ createReactComponent('nexus-stepper');
export const NexusStepperItem = /*@__PURE__*/ createReactComponent('nexus-stepper-item');
export const NexusTabBar = /*@__PURE__*/ createReactComponent('nexus-tab-bar');
export const NexusTable = /*@__PURE__*/ createReactComponent('nexus-table');
export const NexusTextarea = /*@__PURE__*/ createReactComponent('nexus-textarea');
export const NexusToast = /*@__PURE__*/ createReactComponent('nexus-toast');
export const NexusToggle = /*@__PURE__*/ createReactComponent('nexus-toggle');
export const NexusTooltip = /*@__PURE__*/ createReactComponent('nexus-tooltip');
export const NexusTooltipContent = /*@__PURE__*/ createReactComponent('nexus-tooltip-content');
export const NexusTooltipTrigger = /*@__PURE__*/ createReactComponent('nexus-tooltip-trigger');
export const NexusTree = /*@__PURE__*/ createReactComponent('nexus-tree');
export const NexusTreeContent = /*@__PURE__*/ createReactComponent('nexus-tree-content');
export const NexusTreeTrigger = /*@__PURE__*/ createReactComponent('nexus-tree-trigger');
